cpp/src/arrow/dataset/file_parquet_test.cc

```diff
@@ -229,6 +236,57 @@ TEST_F(TestParquetFileFormat, WriteRecordBatchReaderCustomOptions) {
                     *actual_schema);
 }
 
+TEST_F(TestParquetFileFormat, WriteRecordBatchReaderTimeIsAdjustedToUTCFalse) {
```

Comment:
Regarding the `WriteRecordBatchReaderTimeIsAdjustedToUTC` function, could we consider removing some of the duplication?